packages/scanner/src/lib/eventAggregator.ts

```diff
@@ -0,0 +1,39 @@
+import { EventEmitter } from 'events';
+
+export type PendingEvent<E> = {
+  emitter: EventEmitter;
+  event: string;
+  args: E[];
+};
+
+export const MaxMSBetween = 60 * 1000;
+
+// TODO: Unify with the code in packages/cli - find a way to make a common import.
+export default class EventAggregator<E> {
+  constructor(
+    private callback: (events: PendingEvent<E>[]) => void,
+    private maxMsBetween = MaxMSBetween
+  ) {}
+
+  private pending: PendingEvent<E>[] = [];
+  private push(emitter: EventEmitter, event: string, args: E[]) {
+    this.pending.push({ emitter, event, args });
+    this.refresh();
+  }
+
+  private timeout?: NodeJS.Timeout;
+  private refresh() {
+    if (this.timeout) clearTimeout(this.timeout);
+    this.timeout = setTimeout(this.emitPending.bind(this), this.maxMsBetween);
```

Comment:
I just learned that timeouts can be refreshed with `timeout.refresh()` without recreating them. Could be useful especially if there are a lot of events.